File: aoj/volume100/csharp/A10017.cs
using System;

class AOJ
{
	static void Main()
	{
		while (true)
		{
			var input = Console.ReadLine().Split();
			var n = int.Parse(input[0]);
			var x = int.Parse(input[1]);
			if (n == 0 && x == 0)
				break;

			var cnt = 0;
			for (var i1 = 1;    i1 <= n; i1++)
			for (var i2 = i1+1; i2 <= n; i2++)
			for (var i3 = i2+1; i3 <= n; i3++)
				if (i1 + i2 + i3 == x)
					cnt++;
			Console.WriteLine(cnt);
		}
	}
}

File: aoj/volume100/scala/A10017.scala
object Main {
  def main(args: Array[String]) {
    loop
  }

  def loop {
    val input = scala.io.StdIn.readLine split " "
    val n = input(0).toInt
    val x = input(1).toInt
    if (n == 0 && x == 0) {
      return
    }

    var n_ways = 0
    for (v1 <- 1    to n)
    for (v2 <- v1+1 to n)
    for (v3 <- v2+1 to n)
      if (v1 + v2 + v3 == x)
        n_ways += 1
    println(n_ways)
    loop
  }
}
